### PR TITLE
Towards #3204 -FeatureSelection

### PR DESCRIPTION
**Pull request by sfilipi**
Adhering to the #3204 template for the FeatureSelection estimator and transformer.

### CODE REVIEW

File: src/Microsoft.ML.Transforms/CountFeatureSelection.cs [hunk 1]
```diff
@@ -18,7 +18,47 @@
 
 namespace Microsoft.ML.Transforms
 {
-    /// <include file='doc.xml' path='doc/members/member[@name="CountFeatureSelection"]' />
+    /// <summary>
+    ///  Selects the slots for which the count of non-default values is greater than or equal to a threshold.
```
Comment (shmoradims) [inline src/Microsoft.ML.Transforms/CountFeatureSelection.cs]: >default values [](start = 54, length = 14)

create new issue for later: we need to specify for each input type what's the default value. for e.g. it's not clear if for text, default is null, empty string, or whitespaces. same goes with key type.

File: src/Microsoft.ML.Transforms/MutualInformationFeatureSelection.cs [hunk 2]
```diff
@@ -19,7 +19,54 @@
...
+    /// |False|0,7,0 |
+    ///
+    /// This is how the dataset above would look, after fitting the estimator, and transforming the data with the resulting transformer:
+    ///
+    /// | Label |  Features |
+    /// | -- | -- |
+    /// |True |4,0 |
+    /// |False|0,5 |
+    /// |True |4,0 |
+    /// |False|0,5 |
```
Comment (shmoradims) [inline src/Microsoft.ML.Transforms/MutualInformationFeatureSelection.cs]: do we need these examples? we already have similar samples.

File: src/Microsoft.ML.Transforms/CountFeatureSelection.cs [hunk 3]
```diff
@@ -18,7 +18,47 @@
...
+    /// | Output column data type | Same as the input column.|
+    ///
+    /// This transform uses a set of aggregators to count the number of non-default values for each slot(vector element) and
+    /// instantiates a <xref:Microsoft.ML.Transforms.SlotsDroppingTransformer> to actually drop the slots that have the default values for the types
+    /// keeping at least 'count' non-default values.
+    /// This transform is useful when applied together with a <xref:Microsoft.ML.Transforms.OneHotHashEncodingTransformer>.
+    /// The count feature selection can remove the features generated by the hash transform that have no data in the examples.
+    ///
+    /// For example, if we set the count to 3 and fit the estimator, apply the transformer to the following Features column,
+    /// we would see the second solot, containing: NaN, 5, 5, NaN values being dropped because that slot has only two non-default values, the two 5 values.
```
Comment (shmoradims) [inline src/Microsoft.ML.Transforms/CountFeatureSelection.cs]: >solot [](start = 32, length = 5)

slot

please use spell checker. we won't catch all the typos manually.
[https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellChecker](https://marketplace.visualstudio.com/items?itemName=EWoodruff.VisualStudioSpellChecker)

File: src/Microsoft.ML.Transforms/MutualInformationFeatureSelection.cs (hunk 2 above)
Comment (sfilipi) [reply src/Microsoft.ML.Transforms/MutualInformationFeatureSelection.cs]: it is really hard to understand without them, i think. 
The samples should be for how to use it, not to understand what is it doing. @natke what do you think?

Comment (natke) [reply src/Microsoft.ML.Transforms/MutualInformationFeatureSelection.cs]: I like the samples. It helped me understand when I read it.

File: src/Microsoft.ML.Transforms/CountFeatureSelection.cs (hunk 1 above)
Comment (sfilipi) [reply src/Microsoft.ML.Transforms/CountFeatureSelection.cs]: Created: [https://github.com/dotnet/machinelearning/issues/3443](https://github.com/dotnet/machinelearning/issues/3443)

---
In reply to: [277027260](https://github.com/dotnet/machinelearning/pull/3424#discussion_r277027260) [](ancestors = 277027260)

**Pull request merged**